### PR TITLE
Speed up person_distinct_id queries for select clients

### PR DESCRIPTION
**Pull request by macobo**
## Changes

Depends on: https://github.com/PostHog/posthog/pull/7576

This PR switches out person_distinct_id query in most cases with a more optimized one based on an environment variable per-team. This will help solve https://github.com/PostHog/posthog/issues/7548

## How did you test this code?

Existing code should still work as needed. Check out benchmarks.

Benchmark changes:

```
      4849.5±50       4459.0±8.2     0.92  benchmarks.QuerySuite.track_trends_dau
      6649.5±96        6248.5±44     0.94  benchmarks.QuerySuite.track_trends_dau_person_property_filter
      5814.0±42        5272.5±54     0.91  benchmarks.QuerySuite.track_trends_dau_person_property_filter_materialized
```

Wins will be much bigger on cloud due to benchmark dataset being a lot smaller than the client running into trouble has.

### CODE REVIEW

File: ee/clickhouse/models/cohort.py [hunk 1]
```diff
@@ -276,7 +281,11 @@ def recalculate_cohortpeople(cohort: Cohort):
         size_before=before_count[0][0],
     )
 
-    cohort_filter = GET_PERSON_IDS_BY_FILTER.format(distinct_query="AND " + cohort_filter, query="")
+    cohort_filter = GET_PERSON_IDS_BY_FILTER.format(
+        distinct_query="AND " + cohort_filter,
+        query="",
+        GET_TEAM_PERSON_DISTINCT_IDS=get_team_distinct_ids_query(cohort.team_id),
```
Comment (neilkakkar) [inline ee/clickhouse/models/cohort.py:287]: Does this work as you'd expect?

The format happening here: https://github.com/PostHog/posthog/blob/master/ee/clickhouse/sql/person.py#L213 seems to supercede what you're doing here^^

Comment (macobo) [reply ee/clickhouse/models/cohort.py:287]: It should - I updated this at https://github.com/PostHog/posthog/pull/7577/files#diff-8c1a8fdf464c1e77c60c9d3a113fa05e04ba342aa680324f9d037e936f3dfd6cR222

Comment (neilkakkar) [reply ee/clickhouse/models/cohort.py:287]: nvm, missed the update there

File: ee/clickhouse/models/property.py [hunk 2]
```diff
@@ -78,7 +82,9 @@ def parse_prop_clauses(
             else:
                 final.append(
                     "AND {table_name}distinct_id IN ({filter_query})".format(
-                        filter_query=GET_DISTINCT_IDS_BY_PROPERTY_SQL.format(filters=filter_query),
+                        filter_query=GET_DISTINCT_IDS_BY_PROPERTY_SQL.format(
+                            filters=filter_query, GET_TEAM_PERSON_DISTINCT_IDS=GET_TEAM_PERSON_DISTINCT_IDS
```
[the remark this answers is not in the document]
Comment (macobo) [reply ee/clickhouse/models/property.py:86]: I don't have access to team_id in this function and it's called all over the place. Ideally we'd like to, but this particular case should be hit only in lifecycle at the moment so not urgent.

File: ee/clickhouse/queries/trends/breakdown.py [hunk 3]
```diff
@@ -23,7 +24,6 @@
     parse_timestamps,
 )
 from ee.clickhouse.sql.events import EVENT_JOIN_PERSON_SQL
```
Comment (neilkakkar) [inline ee/clickhouse/queries/trends/breakdown.py:26]: rm

**Review by neilkakkar — APPROVED** [review]
Minute benchmark changes are a bit suspicious, but I expect that to be a function of benchmark DB not being too huge to realise these gains.

**Pull request merged**